### ISSUE TITLE
IntelliSense crash with the seqan3 library

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#8956]
I have the same issue.

I used this library:https://github.com/seqan/seqan3

When I include this header, send crashes: 

> #include <seqan3/alignment/pairwise/align_pairwise.hpp>

The following is the call stack
```
Thread 17 "cpptools-srv" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f8742ffd700 (LWP 44992)]
0x000000000095b636 in push_pack_instantiation(a_pack_expansion_descr*, a_template_param*, a_template_arg*, int, int, int, a_ctws_state*, int*) ()
(gdb) bt
#0  0x000000000095b636 in push_pack_instantiation(a_pack_expansion_descr*, a_template_param*, a_template_arg*, int, int, int, a_ctws_state*, int*) ()
#1  0x000000000095c194 in begin_pack_deduction_context(a_pack_expansion_descr*, a_template_param*, a_template_arg**, a_pack_expansion_stack_entry**) ()
#2  0x00000000009b44a4 in matches_template_arg_list(a_template_arg*, a_template_arg*, a_template_arg**, a_template_param*, unsigned int) ()
#3  0x00000000009969be in matches_template_type_for_class_type(a_type*, a_type*, a_template_arg**, a_template_param*, unsigned int) ()
#4  0x000000000099572a in matches_template_type(a_type*, a_type*, a_template_arg**, a_template_param*, unsigned int) ()
#5  0x000000000093e893 in deduce_from_one_pair(a_type*, a_type*, a_type*, a_type*, a_template_arg**, a_template_param*) ()
#6  0x0000000000941340 in deduce_one_parameter(a_param_type*, a_type*, an_init_component**, a_type*, a_symbol*, a_template_arg**) ()
#7  0x0000000000940091 in determine_function_viability(a_symbol*, a_symbol*, int, a_template_arg*, a_symbol*, a_type*, an_init_component*, int, an_operand*, a_type*, int, int, int, int, int, int, int, int, int, int, an_overload_context, a_candidate_function**, int*, int*, int*, int*) ()
#8  0x000000000091fb10 in try_overloaded_function_match(a_symbol*, int, a_template_arg*, an_init_component*, an_init_component*, int, an_operand*, int, int, int, int, int, int, int, int, int, int, an_overload_context, a_candidate_function**, a_symbol**, int*, int*, int*) ()
#9  0x0000000000920ce2 in select_overloaded_function(a_symbol*, int, a_template_arg*, int, an_operand*, an_init_component*, an_init_component*, int, int, int, int, an_overload_context, a_source_position*, unsigned int, int*, int*, int*, int*, a_symbol**, an_arg_match_summary**) ()
#10 0x000000000093d9a7 in deduce_class_template_args(a_type*, int, int, int, an_init_component*, a_source_position*, a_type**, int*) ()
#11 0x000000000078a7ce in deduce_placeholder_type(int, int, int, int, a_type*, a_type*, int, an_operand*, an_init_component*, a_source_position*, a_type**, a_type**, int*) ()
#12 0x000000000078afba in prescan_initializer_for_auto_type_deduction(a_decl_parse_state*, int) ()
#13 0x000000000075a04d in initializer(a_decl_parse_state*, a_source_position*, an_id_linkage_kind, int, int*, a_decl_pos_block*) ()
#14 0x0000000000750cd7 in scan_nonmember_declaration(a_decl_parse_state*, a_source_range*) ()
#15 0x000000000075447a in namespace_declaration(a_token_kind_tag*, int, int, a_source_position*, a_symbol**) ()
#16 0x000000000075438d in namespace_declaration(a_token_kind_tag*, int, int, a_source_position*, a_symbol**) ()
#17 0x000000000074bfcb in scan_nonmember_declaration(a_decl_parse_state*, a_source_range*) ()
#18 0x00000000007521e8 in translation_unit() ()
#19 0x00000000009cf7f4 in process_translation_unit(char const*, int, an_exported_template_file*) ()
#20 0x00000000006a73b7 in cfe_main(int, char**) ()
#21 0x00000000009f3e36 in cfe_main_exception_handler(int, char**) ()
#22 0x00000000006a74b6 in edg_main(int, char**) ()
#23 0x00000000009ed1cc in edge_compiler_main(int, char const**) ()
#24 0x00000000006333b6 in preparse(int, char const**, a_scout_store*, edge::translation_unit*) ()
#25 0x00000000005f87ef in a_compiler_thread::compiler_thread_routine(a_compiler_thread*) ()
#26 0x0000000000adbb0a in msvc::thread_helper_t::thread_entry(void*) ()
#27 0x00007f8753217609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#28 0x00007f875313e293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

I'm a beginner, and I can't help you with debugging such a complex project. I hope the information I provide can help you locate the problem.

_Originally posted by @66maer in https://github.com/microsoft/vscode-cpptools/issues/8678#issuecomment-1055410448_

**Comment by sean-mcmanus** [comment]
I've verified the repro on Linux with g++ and c++20.

**Comment by sean-mcmanus** [comment]
I've filed an internal bug against our shared VS parser (1491631).

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.3 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.3

**Comment by 66maer** [comment]
自动回复：我已经收到您的邮件，谢谢。

**Comment by sean-mcmanus** [comment]
This fix is being removed in 1.13.4 due to regression https://github.com/microsoft/vscode-cpptools/issues/10075 . Once that gets fixed we can add back the fix.

**Comment by sean-mcmanus** [comment]
The fix is added back in 1.14.0 (pre-release): https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0

**Comment by 66maer** [comment]
自动回复：我已经收到您的邮件，谢谢。
